plugins/redis/redis-cli.go

```diff
@@ -0,0 +1,34 @@
+package redis
+
+import (
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/needsauth"
+	"github.com/1Password/shell-plugins/sdk/schema"
+)
+
+func RedisCLI() schema.Executable {
+	return schema.Executable{
+		Name:    "Redis CLI",
+		Runs:    []string{"redis-cli"},
+		DocsURL: sdk.URL("https://redis.io/docs/ui/cli"),
+		NeedsAuth: needsauth.IfAll(
+			needsauth.NotWhenContainsArgs("-u"),
+			needsauth.NotWhenContainsArgs("--user"),
+			needsauth.NotWhenContainsArgs("-a"),
+			needsauth.NotWhenContainsArgs("--help"),
+			needsauth.NotForVersion(),
+		),
+		Uses: []schema.CredentialUsage{
```

Comment:
No need to use `SelectFrom` here. Let's just add the credential name `User Credentials`.  The reason for this is that redis-cli can only work with the user credentials. No other credentials have any significance for the redis cli executable.